lib/puppetserver/ca/utils/http_client.rb

```diff
@@ -196,7 +196,13 @@ def self.check_server_online(settings, logger)
             if e.wrapped.is_a? Errno::ECONNREFUSED
               return false
             else
-              raise e
+              if suppress_error
```

Comment:
might want to smush this into an `elsif`? I find that easier to read than nested conditionals.